test/unit/CMakeLists.txt

```diff
@@ -52,7 +52,8 @@ unset(_save_cxx_flags)
 #
 add_executable(proj_test_unit
   main.cpp
-  basic_test.cpp)
+  basic_test.cpp
+  proj_errno_string_test.cpp)
```

Comment:
I strongly encourage you to not combine compilation units into the same test target.  You now must make absolutely sure that proj_errno_string_test.cpp leaves any global in a state that exactly matches as if the tests were never run.  If you fail to cleanup or have a test barf, you risk making the problem much harder to follow.  Also, if you separate them out, you can run them in parallel in separate processes... I need to figure out how to parallelize that.  I'm guess that this doesn't parallelize the unit tests :(

```
rm -rf build-ninja; (mkdir -p build-ninja && cd build-ninja && cmake -GNinja .. && cmake --build . && ctest .)
```

Reply:
Okay, I didn't realise what I was doing here. I'll change this to a separate test target.

Reply:
There will be a significant cost in term of linking time if we add a binary for each test we write. For my C++ SRS stuff, I intend to have just one test binary. Normally PROJ shouldn't cause global side effects, except perhaps the global default context (but in that case we should have a way to reset it between runs)

Reply:
I've not done testing at scale with cmake or autoconf much, so I differ to @rouault .  Go with one test bin and if the world gets too painful, we can pull out a key test file or two to separate binaries.  I just want to take advantage of the large number of cores that workstations are starting to have (I've got 28 physical cores for 56 hyperthreads), but I know that the web based CI systems will go really slow if link time is large.

Reply:
I am indifferent at the moment - I'll just go with whatever you guys decide is best. I might come back with a stronger opinion after having played around with the test framework for a while.